sale_commission_product_criteria/models/sale.py

```diff
@@ -0,0 +1,54 @@
+# © 2023 ooops404
+# License AGPL-3 - See https://www.gnu.org/licenses/agpl-3.0.html
+from odoo import api, fields, models
+
+
+class SaleOrderLineAgent(models.Model):
+    _inherit = "sale.order.line.agent"
+
+    discount = fields.Float(related="object_id.discount")
+    applied_commission_item_id = fields.Many2one("commission.item")
+    based_on = fields.Selection(related="applied_commission_item_id.based_on")
+    applied_on_name = fields.Char(related="applied_commission_item_id.name")
+    commission_type = fields.Selection(
+        related="applied_commission_item_id.commission_type"
+    )
+    fixed_amount = fields.Float(related="applied_commission_item_id.fixed_amount")
+    percent_amount = fields.Float(related="applied_commission_item_id.percent_amount")
+
+    @api.depends(
+        "object_id.price_subtotal", "object_id.product_id", "object_id.product_uom_qty"
+    )
+    def _compute_amount(self):
+        res = None
+        for line in self:
+            if line.commission_id and line.commission_id.commission_type == "product":
+                order_line = line.object_id
+                line.amount = line._get_single_commission_amount(
+                    line.commission_id,
+                    order_line.price_subtotal,
+                    order_line.product_id,
+                    order_line.product_uom_qty,
+                )
+            else:
+                res = super(SaleOrderLineAgent, line)._compute_amount()
+        return res
```

Comment:
```suggestion
    def _compute_amount(self):
        lines = self.env["sale.order.line.agent"]
        for line in self:
            if line.commission_id and line.commission_id.commission_type == "product":
                order_line = line.object_id
                line.amount = line._get_single_commission_amount(
                    line.commission_id,
                    order_line.price_subtotal,
                    order_line.product_id,
                    order_line.product_uom_qty,
                )
            else:
                lines += line 
        return super(SaleOrderLineAgent, lines)._compute_amount()
```
I believe the most appropriate way to achieve the goal is the following

Reply:
Why this change is not set?